src/SocialPostManager.php

```diff
@@ -0,0 +1,280 @@
+   * dispatchers.
+   *
+   * @return string
+   *   The plugin id.
+   */
+  public function getPluginId() {
+    return $this->pluginId;
+  }
+
+  /**
+   * Add user record in Social Post Entity.
+   *
+   * @param string $pluginId
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   * @param string $token
+   *   Token to be used for autoposting.
+   *
+   * @return True
+   *   if User record was created or
+   *   False otherwise
+   */
+  public function addRecord($pluginId, $provider_user_id, $token) {
+    // Get User ID of logged in user.
+    $user_id = $this->currentUser->id();
+
+    // Encode token into json format.
+    $json_token = json_encode($token);
+
+    // Add user record.
+    $values = [
+      'user_id' => $user_id,
+      'plugin_id' => $pluginId,
+      'provider_user_id' => $provider_user_id,
+      'token' => $this->encryptToken($json_token),
+    ];
+
+    $user_info = $this->entityTypeManager->getStorage('social_post')->create($values);
+    if ($user_info) {
+      return TRUE;
+    }
+    else {
+      return FALSE;
+    }
+
+  }
+
+  /**
+   * Update token of a particular record.
+   *
+   * @param string $pluginId
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   * @param string $token
+   *   Token provided by social_network.
+   *
+   * @return True
+   *   if updated else
+   *   False otherwise.
+   */
+  public function updateToken($pluginId, $provider_user_id, $token) {
+    $field_storage_configs = $this->entityTypeManager
+      ->getStorage('social_post')
+      ->loadByProperties(['plugin_id' => $pluginId, 'provider_user_id' => $provider_user_id]);
+
+    foreach ($field_storage_configs as $field_storage) {
+      $field_storage->token = $token;
+      $field_storage->enforceIsNew(FALSE);
+      $save_token = $field_storage->save();
+    }
+
+    if ($save_token) {
+      return TRUE;
+    }
+    else {
+      return FALSE;
+    }
+  }
+
+  /**
+   * Delete record from entity table.
+   *
+   * @param string $pluginId
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   *
+   * @return True
+   *   if deleted else
+   *   FALSE otherwise.
+   */
+  public function deleteRecord($pluginId, $provider_user_id) {
+    $delete_record = $this->entityTypeManager
+      ->getStorage('social_post')
+      ->loadByProperties(['plugin_id' => $pluginId, 'provider_user_id' => $provider_user_id])
+      ->delete();
+
+    if ($delete_record) {
+      return TRUE;
+    }
+    else {
+      return FALSE;
+    }
+  }
+
+  /**
+   * Used to get token for autoposting by implementers.
+   *
+   * @param string $pluginId
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   *
+   * @return Token
```

Comment:
The return statement should be something like
[at]return string
  The token in array format

The type after @return serves to IDE to know what kind of object is expected expect.